shell/platform/darwin/ios/framework/Source/FlutterEngineGroup.mm

```diff
@@ -0,0 +1,60 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Headers/FlutterEngineGroup.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterEngine_Internal.h"
+
+@interface FlutterEngineGroup ()
+@property(nonatomic, copy) NSString* name;
+@property(nonatomic, strong) NSMutableArray<NSValue*>* engines;
+@property(nonatomic, strong) FlutterDartProject* project;
+@end
+
+@implementation FlutterEngineGroup {
+  int _engineCount;
+}
+
+- (instancetype)initWithName:(NSString*)name project:(nullable FlutterDartProject*)project {
+  self = [super init];
+  if (self) {
+    self.name = name;
```

Comment:
let's engineer solutions that need solving now. It seems like this is just going to be used for naming threads in the threadhost for instruments. Except we're only making one group of threads. 

Maybe just scope this to mean the thread name of the first engine/thread?

Reply:
addressed in "xiao feedback 1" -- Added initWithName:project: docstring.